src/uct/cuda/gdr_copy/gdr_copy_md.c

```diff
@@ -251,22 +251,50 @@ uct_gdr_copy_mem_dereg(uct_md_h uct_md,
     return status;
 }
 
+static int uct_gdr_is_gdr_get_info_v1(int major, int minor)
+{
+    return (major < 2) || ((major == 2) && (minor <= 3));
+}
+
+static int uct_gdr_is_gdr_compatible(void)
+{
+    int major, minor;
+
+    gdr_runtime_get_version(&major, &minor);
+    if (uct_gdr_is_gdr_get_info_v1(GDR_API_MAJOR_VERSION,
+                                   GDR_API_MINOR_VERSION) !=
+        uct_gdr_is_gdr_get_info_v1(major, minor)) {
+        ucs_warn("could not open gdr copy, built for v%d.%d but loaded v%d.%d",
+                 GDR_API_MAJOR_VERSION, GDR_API_MINOR_VERSION, major, minor);
```

Comment:
i'd move this warning to uct_gdr_copy_query_md_resources since that function prints the warning in other error cases